### PR TITLE
Add Speech Samples

### PR DESCRIPTION
**Pull request by ahmedalmoraly**
Restructure the repo to add speech samples

### CODE REVIEW

**Comment by ahmedalmoraly** [comment]
Updated to use the new `affdex::vision` API

File: speech/README.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+See http://www.portaudio.com/
+
+* For CentOS : ` yum install -y portaudio-devel.x86_64`
+
+#### LibSndFile: for building the `wav` sample
+
+See http://www.mega-nerd.com/libsndfile/
+
+* For CentOS : `yum install -y libsndfile-devel.x86_64`
+
```
Comment (umangmehta12) [inline speech/README.md]: For Ubuntu: `sudo apt-get install libsndfile1-dev`

File: speech/README.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+See http://www.boost.org/
+
+
+#### PortAudio: for building the `mic` sample
+
+See http://www.portaudio.com/
+
+* For CentOS : ` yum install -y portaudio-devel.x86_64`
+
```
Comment (umangmehta12) [inline speech/README.md]: For Ubuntu: `sudo apt-get install libportaudio2`

File: vision/README.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+- Boost
+
+```
+$ wget https://sourceforge.net/projects/boost/files/boost/1.63.0/boost_1_63_0.tar.gz
+$ tar -xzvf boost_1_63_0.tar.gz -C $HOME
+$ cd boost_1_63_0
+$ ./bootstrap.sh
+$ sudo ./b2 -j $(nproc) cxxflags=-fPIC threading=multi runtime-link=shared \
+      --with-log --with-serialization --with-system --with-date_time \
```
Comment (umangmehta12) [inline vision/README.md]: Even here we should reduce the number of libraries for boost. I can see only the following boost libraries used filesystem, timer, program_options, algorithm, and format. Not sure which of the above libraries install algorithm and format. A quick look at the documentation seems like boost builds (algorithm and format) by default and we have to specify the rest of the libraries(https://www.boost.org/doc/libs/1_66_0/more/getting_started/unix-variants.html).

**Pull request merged**